patchkit-launcher-qt-src/src/lockfile.cpp

```diff
@@ -36,24 +72,54 @@ void LockFile::unlock()
 {
     if (isLockedLocally())
     {
+        system_unlock(m_lockFile.handle());
+        m_lockFile.close();
         m_lockFile.remove();
         m_isLockFileLocal = false;
     }
 }
 
 void LockFile::cede()
 {
+    system_unlock(m_lockFile.handle());
+    m_lockFile.close();
     m_isLockFileLocal = false;
 }
 
 bool LockFile::isLocked() const
 {
-    return m_lockFile.exists();
+    if (!m_lockFile.exists())
+    {
+        qInfo("Lock file doesn't exist - not locked.");
+        return false;
+    }
+
+    if (isLockedLocally())
+    {
+        qWarning("Lock file is locked by this program - locked.");
+        return true;
+    }
+
+    bool lockResult = system_lock(m_lockFile.handle());
```

Comment:
Method name isLocked() is no longer correct if this method is actually locking the file. Consider introducing a new method called tryLock().